tests/validation/tests/v3_api/test_workload.py

```diff
@@ -858,3 +858,333 @@ def setup_project_by_role(role, remove_resource):
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_delete_negative(remove_resource):
+    if role == CLUSTER_OWNER:
+        # cluster owner can edit daemonsets in any project in the cluster
+        project = rbac_get_project()
+        user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        p2 = rbac_get_unshared_project()
+        p_client = get_project_client_for_token(project, user_token)
+        p_client2 = get_project_client_for_token(p2, user_token)
+        ns2 = rbac_get_unshared_ns()
+        name = random_test_name("default")
+        con = [{"name": "test1",
+                "image": TEST_IMAGE}]
+        workload = p_client2.create_workload(name=name,
+                                    containers=con,
+                                    namespaceId=ns.id,
+                                    daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        con = [{"name": "test1", "image": "nginx"}]
+        p_client.update(workload, containers=con)
+        validate_workload_image(p_client, workload, "nginx", ns)
+        p_client.delete(workload)
+
+    else:
+        if role == CLUSTER_MEMBER:
+            user_token = rbac_get_user_token_by_role(role)
+            project, ns = create_project_and_ns(user_token, namespace["cluster"],
+                                                random_test_name("test-rbac"))
+            p_client = get_project_client_for_token(project, user_token)
+        else:
+            project = rbac_get_project()
+            user_token = rbac_get_user_token_by_role(role)
+            p_client = get_project_client_for_token(project, user_token)
+            ns = rbac_get_namespace()
+        # roles cannot delete daemonsets in projects they cannot access
+        # deploy a workload as cluster owner
+        cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        cluster_owner_p_client = get_project_client_for_token(project,
+                                                            cluster_owner_token)
+        ns = rbac_get_namespace()
+        con = [{"name": "test1", "image": TEST_IMAGE}]
+        name = random_test_name("default")
+        workload = cluster_owner_p_client.create_workload(name=name,
+                                                        containers=con,
+                                                        namespaceId=ns.id
+                                                        daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        # project read-only can NOT delete existing workload
+        with pytest.raises(ApiError) as e:
+            p_client.delete(workload)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'Forbidden'
+
+        remove_resource(workload)
```

Comment:
flake8 is complaining mostly of visual indent or lines too long:

`flake8 tests/v3_api/test_workload.py --ignore=F405,F401,F811`

```
tests/v3_api/test_workload.py:918:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:919:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:920:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:934:41: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:935:41: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:936:41: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:978:31: E999 SyntaxError: invalid syntax
tests/v3_api/test_workload.py:1006:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1007:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1008:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1019:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1020:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1021:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1048:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1049:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1050:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1060:80: E501 line too long (81 > 79 characters)
tests/v3_api/test_workload.py:1072:61: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1072:80: E501 line too long (80 > 79 characters)
tests/v3_api/test_workload.py:1077:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1078:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1079:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1151:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1152:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1153:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1163:80: E501 line too long (81 > 79 characters)
tests/v3_api/test_workload.py:1175:61: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1175:80: E501 line too long (80 > 79 characters)
tests/v3_api/test_workload.py:1180:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1181:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1182:57: E128 continuation line under-indented for visual indent
```

Reply:
`flake8` is still unhappy.